wolfcrypt/src/evp.c

```diff
@@ -5459,9 +5658,23 @@ void wolfSSL_EVP_init(void)
                 case AES_256_XTS_TYPE:
     #endif
                     wc_AesFree(&ctx->cipher.aes);
+                    break;
+#endif /* AES */
+    #ifdef HAVE_ARIA
+                case ARIA_128_GCM_TYPE:
+                case ARIA_192_GCM_TYPE:
+                case ARIA_256_GCM_TYPE:
+                    {
+                        int ret = wc_AriaFreeCrypt(&ctx->cipher.aria);
+                        if (ret != 0) {
+                            WOLFSSL_MSG("wc_AriaFreeCrypt failure");
+                            return WOLFSSL_FAILURE;
```

Comment:
A failure here will leak all the cleanup below. Consider logging, but not returning here.